packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -48,7 +50,7 @@ export default util.createRule<Options, 'floating'>({
 
         if (isUnhandledPromise(checker, expression)) {
           context.report({
-            messageId: 'floating',
+            messageId: options.ignoreVoid ? 'floatingVoid' : 'floating',
```

Comment:
This would probably be better as a suggestion fixer, to make it easy to fix to.

```suggestion
            messageId: options.ignoreVoid ? 'floatingVoid' : 'floating',
            suggest: options.ignoreVoid ? [
              {
                messageId: 'xxx', // Consider explicitly marking
                fix(fixer) { /* ... */ }
              },
            ] : [],
```

Reply:
But won't that autofix with the `--fix` eslint argument? Or `eslint.fixAll` code action in vscode? That would suck because I think it should be the developer's conscious decision.

Reply:
Nope, that's the beauty of eslint suggestions!

For fixers, there is exactly 0 or 1 fixers for each report. A fixer _will_ be applied by eslint's `--fix` mode, if it exists for a report.

For suggestion fixers, on the other hand, can have 0..n suggestions for each report. Each suggestion has its own message which is used to communicate what it does to the user. Each suggestion is not automatically applied by eslint's `--fix` mode.

Suggestions were added relatively recently, and are great for doing these sorts of fixes - ones that might break code, or require conscious decisions from the user. And the fact that you can present multiple for a report means that you can present the user with options for correcting the error.

Reply:
Done!